modules/videoio/src/cap.cpp

```diff
@@ -112,10 +112,13 @@ bool VideoCapture::open(const String& filename, int apiPreference)
                             CV_LOG_WARNING(NULL, cv::format("VIDEOIO(%s): can't create capture", info.name));
                     }
                 } catch(const cv::Exception& e) {
+                    if(errorOnFail) throw;
```

Comment:
This function iterates over all available backends, so re-throwing exceptions from here break normal control flow.
For example, FFmpeg backend can't propertly handle "/dev/video0", but V4L or GStreamer can.

Probably it make sense only if `apiPreference != CAP_ANY`.